lib/galaxy/schema/schema.py

```diff
@@ -1331,42 +1369,24 @@ class HistoryActiveContentCounts(Model):
 HistoryContentStateCounts = Dict[HistoryContentStates, int]
 
 
-class HistoryDetailed(HistorySummary):  # Equivalent to 'dev-detailed' view, which seems the default
+class HistoryDetailed(HistoryCommonDetailed):
     """History detailed information."""
 
-    contents_url: ContentsUrlField
-    size: int = Field(
-        ...,
-        title="Size",
-        description="The total size of the contents of this history in bytes.",
-    )
-    user_id: Optional[EncodedDatabaseIdField] = Field(
+    email_hash: Optional[str] = Field(
         None,
-        title="User ID",
-        description="The encoded ID of the user that owns this History.",
+        title="Encoded Email",
+        description="Encoded owner email.",
     )
-    create_time: datetime = CreateTimeField
-    importable: bool = Field(
+    empty: bool = Field(
```

Comment:
This seems wrong. This is quite redundant with the hid_counter or whatever we call it.

Reply:
I agree, there are a bunch of redundant properties also like `nice_size` or `type_id` (both can be computed in the client) or `file_ext` and `extension` for datasets.

This `empty` seems to have been there since 2014 https://github.com/galaxyproject/galaxy/blame/6b96bb17b2099c527fd5e6b02699e8f6b81b6593/lib/galaxy/managers/histories.py#L793 on the other hand, what we really are using for detailed views in the client is this `dev-detailed` view (and not the `detailed`) because most of the requests for details are for the current history with `history/current_history_json`.

Again we should really try to untangle this serialization and fine-tune the models.

Should we do something about this in this PR?

Reply:
> because most of the requests for details are for the current history with history/current_history_json.

what a mess :(. 

> Should we do something about this in this PR?

can we get away with not changing the models ?

Reply:
Yes, sure! I already felt there were too many changes for a bug fix, but I wanted to "accurately expose" what we are "really" retrieving from the server and storing in the client.

I can revert the API model changes in the backend and try to mimic them in the client to achieve something similar. Then we can follow up on dev trying to migrate this `history/current_history_json` and associates to the API and fine-tune the models at that point if we can.

Sounds good?

Reply:
Yes, thank you!

Reply:
Done! I've documented in the client schemas `client/src/api/index.ts` the inconsistencies with the views and what we usually request from the client and serialize from the backend.